bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngineImpl.java

```diff
@@ -589,9 +584,10 @@ protected void unsetEventPublisher(EventPublisher eventPublisher) {
      */
     protected void postRuleStatusInfoEvent(String ruleUID, RuleStatusInfo statusInfo) {
         if (eventPublisher != null) {
+            EventPublisher ep = eventPublisher;
```

Comment:
why? also line 590.

Reply:
To avoid (valid) NPE compiler warnings. Field could be set to null while we are executing this code block.